src/main/java/seedu/address/ui/InfoPanel.java

```diff
@@ -68,17 +68,14 @@
      * Creates a {@code InfoPanel} with the given {@code Person}.
      * @param p
      */
-    public InfoPanel(ObservableObjectValue<Person> p) {
+    public InfoPanel(ObservablePerson p) {
         super(FXML);
 
         textOverlay.setText(LIST_EMPTY_MSG);
         textOverlay.setTextAlignment(TextAlignment.CENTER);
 
         updateInfoPanel(p.get());
-        p.addListener((x, y, z) -> {
-            updateInfoPanel(z);
-            System.out.println(y);
-        });
+        p.addUiObserver(this);
```

Comment:
Just a really small nitpick, but maybe it would be clearer to rename the parameter here to something clearer like 'viewingPerson'?

Reply:
Not really sure which parameter your referring to, but added comments to try to explain the line.